frontend/src/stimulus/controllers/external-links.controller.ts

```diff
@@ -29,61 +29,84 @@
 
 /**
- * Observes all external links and sets their ARIA `describedby` attribute to
- * {BLANK_LINK_DESCRIPTION_ID} - this element should exist in the DOM and
- * provide localized text content along the lines of "Open link in a new tab".
+ * Dynamically observes all links in the page, including those added later via Turbo frames or DOM mutations.
  *
- * The goal is to make users of Assistive Technology aware that they may have to
- * switch tabs on clicking a link.
+ * For external links (pointing to a different domain than the current page):
+ *   - Sets `target="_blank"` to open in a new tab.
+ *   - Sets `rel="noopener noreferrer"` for security and performance.
+ *   - Adds `aria-describedby` pointing to a description element (`BLANK_LINK_DESCRIPTION_ID`) to inform
+ *     users of assistive technologies that the link opens in a new tab.
  *
- * We consider links with a `target` attribute set to "_blank" as "external".
+ * For internal links (same domain):
+ *   - Ensures `target="_top"` and removes the `rel` attribute to keep default behavior.
+ *
+ * This ensures accessibility, security, and consistent behavior for all links, including dynamically
+ * loaded content.
  */
 export default class ExternalLinksController extends ApplicationController {
   connect() {
-    useMutation(this, { attributes: true, childList: true, subtree: true, attributeFilter: ['target'] });
+    useMutation(this, { attributes: true, childList: true, subtree: true, attributeFilter: ['href'] });
 
-    // initial pass
-    document.querySelectorAll(BLANK_LINK_QUERY).forEach(applyLinkDescription);
+    // Initial pass: handle existing blank links (accessibility)
+    document.querySelectorAll<HTMLAnchorElement>(EXTERNAL_LINK_QUERY).forEach((link)=>{
+      if (link.hostname !== window.location.hostname) {
+        updateExternalLink(link);
+      }
+    });
   }
 
   mutate(mutations:MutationRecord[]) {
     mutations.forEach((mutation) => {
       mutation.addedNodes.forEach((node) => {
         if (isElement(node)) {
-          // added element itself is a blank link
-          if (isBlankLink(node)) {
-            applyLinkDescription(node);
-          }
-          // added sub-trees
-          node.querySelectorAll(BLANK_LINK_QUERY).forEach(applyLinkDescription);
+          // Added element itself is a blank link
+          if (isExternalLink(node)) updateExternalLink(node);
+
+          // Added sub-trees
+          node.querySelectorAll<HTMLAnchorElement>(EXTERNAL_LINK_QUERY).forEach((link)=>{
+            if (link.hostname !== window.location.hostname) {
+              updateExternalLink(link);
+            }
+          });
         }
       });
 
-      // attribute changes
+      // Attribute changes
       if (
         mutation.type === 'attributes' &&
         mutation.attributeName === 'target' &&
```

Comment:
We need to test for the `href` attribute here, not `target`:

```suggestion
        mutation.attributeName === 'href' &&
```